crates/bevy_reflect/src/lib.rs

```diff
@@ -2924,10 +2924,6 @@ bevy_reflect::tests::Test {
 
     #[test]
     fn should_reflect_external_crate_type() {
-        // This test relies on the external type not implementing `Reflect`,
-        // so let's just double-check that it does not
-        assert_not_impl_all!(std::collections::Bound<i32>: Reflect);
```

Comment:
This test is actually testing that remote reflection works for a real external crate (`std` in this case). We should either replace the type reflected in this test or remove the test entirely.

The latter is fine since we have other tests, but the former is probably ideal.

Reply:
I think since the latter is easier and reducing test times is a nice bonus, I will do that.

Reply:
Actually I did not see that the former was ideal.... I wonder, do we need to have a type that doesn't implement reflect but is in std? it seems like that would set us up to have to rewrite the test though, unless we can identify something that will never implement reflect, which IMO is hard to make a case for. Just curious on whether that's the case or not.

Reply:
> Actually I did not see that the former was ideal.... I wonder, do we need to have a type that doesn't implement reflect but is in std? 

It doesn't have to be in `std`, that's just what I used since it was easy to access. We could use any external type pulled from a dev-dependency, for example.

> it seems like that would set us up to have to rewrite the test though, unless we can identify something that will never implement reflect, which IMO is hard to make a case for. Just curious on whether that's the case or not.

Yeah the test would have to be rewritten to use the new type. Again, I think it's fine if you want to skip doing that since we sorta know it works on external types (no traits are implemented on the remote type by the `#[reflect_remote]` macro).

Reply:
> I think it's fine if you want to skip doing that since we sorta know it works on external types

In that case since I can't seem to find a good replacement type without making a whole other crate just for this test, I think we'll just go with that.